linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/dashboard.js

```diff
@@ -60,3 +69,181 @@ var Dashboard = (function() {
+        routerServers.onMetricsUpdate();
+      },
+      desiredMetrics: function() {
+        return _.concat(
+          routerSummaries.desiredMetrics(),
+          routerServers.desiredMetrics()
+        )
+      }
+    };
+  };
+})();
+
+var RouterServers = (function() {
+  // Skeletal implementation to demonstrate functionality of RouterController
+
+  function initializeServerContainers(servers, routerContainerEls) {
+    _.each(servers, function(server) {
+      var $router = $(routerContainerEls[server.router].find(".server")[0]);
+
+      var $serverEl = $("<div />").addClass("router-server")
+        .html("<div>Servers</div><div>" + server.prefix + "</div>");
+      $serverEl.appendTo($router);
+    });
+  }
+
+  function renderServers() {
+  }
+
+  return function (selectedRouter, routers, routerContainerEls) {
+    var servers = routers.servers(selectedRouter);
+    initializeServerContainers(servers, routerContainerEls);
+
+    return {
+      onMetricsUpdate: function() {
+        renderServers();
+      },
+      desiredMetrics: function () {
+        return [];
+      }
+    }
+  };
+})();
+
+var RouterSummaries = (function() {
+  var baseValues = { load: 0, requests: 0, success: 0, failures: 0 };
+  var summaryKeys = _.keys(baseValues);
+  var metricsParams = [];
+  var summaryEls = {};
+
+  function addClients(clients) {
+    metricsParams = metricsParams.concat(clientsToMetricParam(clients));
+  }
+
+  function clientsToMetricParam(clients) {
+    return _(clients).map(function(client) {
+      return _.map(summaryKeys, function(key) {
+        return client.prefix + key;
+      });
+    }).flatten().value();
+  }
+
+  function processResponses(data) {
+    return _.map(data, function(data, router) {
+      return appendSuccessAndFailureRate(data);
+    });
+  }
+
+  function appendSuccessAndFailureRate(datum) {
+    var result = $.extend({}, baseValues, datum);
+    var successRate = new SuccessRate(result.requests, result.success, result.failures);
+    result.successRate = successRate.prettyRate();
+    result.failureRate = getFailureRate(result);
+
+    return result;
+  }
+
+  function getFailureRate(datum) {
+    // there's some discussion as to whether we should include both success
+    // and failure rate.  this is a very sketchy implementation of this until
+    // we decide for sure
+    if (datum.requests === 0) {
+      return "N/A";
+    } else {
+      return (100*datum.failures/datum.requests).toFixed(2) + "%";
+    }
+  }
+
+  function renderRouterSummary(routerData, template, routerEls) {
+    _.each(routerData, function(stats) {
+      var summaryHtml = template(stats);
+
+      if (!summaryEls[stats.router]) {
+        var $summary = $("<div />").html(summaryHtml);
+        var $router = $(routerEls[stats.router].find(".summary")[0]);
+
+        summaryEls[stats.router] = $summary;
+        $summary.appendTo($router);
+      } else {
+        summaryEls[stats.router].html(summaryHtml);
+      }
+    });
```

Comment:
I think this method could be a lot simpler if RouterSummary was responsible for rendering a single summary instead of a list of summaries. In a single summary scenario, the RouterController parent could pass in a $summaryEl to render the template in & there would be no need for the `summaryEls` mapping